### ISSUE TITLE
Have a way to retry failed requests

### ISSUE THREAD

**Issue by csomh** [packit/ogr#699]
In Packit Service, with many API calls going on in parallel for the same repository, even though there is tokman to help with this, it can still happen, that credentials expire and so these requests fail, even though the previous one just succeeded.

It would be nice to have a way to configure retries for given exceptions. For example, to be able to retry for `github.BadCredentialsException`s, 3 times, with some exponential backoff.
